### ISSUE TITLE
pipenv adds Pipfile entry for failed module installs

### ISSUE THREAD

**Issue by jtmoon79** [pypa/pipenv#3523]
***tl;dr*** `pipenv install NON_EXISTENT_MODULE` adds an `[packages]` entry to Pipfile `NON_EXISTENT_MODULE= "*"`.

### Issue description

Pipenv adds an entry for failed installs.

### Expected result

Pipenv shouldn't modify the Pipfile for a failed install.

### Steps to replicate

1. Note contents of `Pipfile` section `[packages]`.
2. Run
`pipenv install NON_EXISTENT_MODULE`
command will fail with much error output and final message
"`ERROR: ERROR: Could not find a version that matches NON_EXISTENT_MODULE`"
3. Note contents of `Pipfile` section `[packages]`.  It now has line
`NON_EXISTENT_MODULE= "*"`

-------------------------------------------------------------------------------

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2018.11.26'`

Pipenv location: `'c:\\python\\python37\\lib\\site-packages\\pipenv'`

Python location: `'c:\\python\\python37\\python.exe'`

Python installations found:

  - `3.7.1`: `C:\Python\Python37\python.exe`
  - `3.6.7`: `C:\Python\Python36\python.exe`
  - `3.5`: `C:\Python\Python35\python.exe`
  - `3.4`: `C:\Python\Python34\python.exe`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.7.1',
 'os_name': 'nt',
 'platform_machine': 'AMD64',
 'platform_python_implementation': 'CPython',
 'platform_release': '10',
 'platform_system': 'Windows',
 'python_full_version': '3.7.1',
 'python_version': '3.7',
 'sys_platform': 'win32'}
```

</details>

**Comment by geoffroy-noel-ddh** [comment]
I think this is a duplicate of this issue: #3389 , which apparently has been fixed in master. So we'll have to wait for next release.

**Comment by joelachance** [comment]
This issue still isn't resolved in either of the two above tickets, can this be re-opened?
